### ISSUE TITLE
Code Audit against v5.01 distribution disks

### ISSUE THREAD

**Issue by the-grue** [roelandjansen/pcmos386v501#47]
Ok, here is my code audit.  I haven't packaged an update yet because I would like folks to review this document beforehand and make sure it is agreed it is the right thing to do.  

The audit compared similar copies of all files across the entire repo, their build objects, file sizes, and checksums of code files and binaries.

Then all of the code was gathered into what appears to be the closest copy of the 5.01 source and built again then compared a second time with file sizes and checksums against the distribution disks.

The document shows the binary file names, if it exists on the disk, if it exists as an output object in the repo, whether or not they match (checksum/size) and if not, where they differ (checksum/size/both).  I also added a notes field.  At the bottom, there is a group of files that are built out of the repo that are not on the distribution disk.  Some may be tools, some may be newer drivers, etc.

Here is the document, please review it and let me know if you have any questions or comments:

[Audit501.xlsx](https://github.com/roelandjansen/pcmos386v501/files/2800564/Audit501.xlsx)

If we are in agreement that this is the right thing to do, I will post a patch with a new directory with all of the gathered files.  From this directory, we could make all of the buildable objects at or as close to the 5.01 distribution as possible.  I have been calling this /SOURCES/src/mos501src, but like everything else, it is open to discussion.

I will follow up with another audited tree that will include any updated code.  Perhaps a /SOURCES/src/latest.

**Comment by src153** [comment]
Can you wait a few days? This is worth studying, but ATM, I can't read a .xlsx file. Don't worry, I'll figure something out. Might be a good idea to let it marinate for some days. Don't want to rush a good meal. The fast food generation may not appreciate that, but I do.

**Comment by the-grue** [comment]
No problem at all.  I can add a text version as well.

I agree it needs some consideration and the more eyes (or ayes as it may be) the better.

Building the latest and putting the finishing touches on that one as well.  Will submit .xlsx and text versions.

**Comment by src153** [comment]
Here'a an idea. Postpone any further work on this until somebody finds and fixes the date bug. Until that happens, any other progress is pointless. This project is going nowhere until the date bug is fixed. Makes me think of the film Boondock Saints where the cop says to the dead thug face down in the alley:

`Where are you going? NOWHERE.`

**Comment by the-grue** [comment]
Plain text is ugly.  Here's a pdf.

[Audit501.pdf](https://github.com/roelandjansen/pcmos386v501/files/2800899/Audit501.pdf)

I think everyone is paused at the moment waiting on this change or a date bug fix.  Since I have been working exclusively outside the existing code tree, this won't make a difference one way or another to any projects folks are currently working on.  As long as we can agree to work off the same set of files.

My submissions would live exclusively outside the existing code directories for both the 501 audited and latest audited files.

**Comment by the-grue** [comment]
Ok, here's the audit of the files with changes between the 5.01 distribution and the end of time as far as the repo is concerned in both .xlsx and .pdf.

[AuditLatest.xlsx](https://github.com/roelandjansen/pcmos386v501/files/2800908/AuditLatest.xlsx)

[AuditLatest.pdf](https://github.com/roelandjansen/pcmos386v501/files/2800910/AuditLatest.pdf)

This includes the filenames, where the newest one was from, (or where it continues to live in a few cases) and any comments in the file related to what may have changed.  Most of the files seem to have cleanup or bug fixes related to updat501 or static patches or random bug fixes.

**Comment by src153** [comment]
I understand yours are outside. At least for now. But once we agree the audited ones are better, the existing should be replaced. Or at least deprecated.

Moses was on the mountain 40 days to get the Ten Commandments. It takes time to write things in stone.

**Comment by src153** [comment]
I looked at 501 and latest. It's a good overview of what we have to work with.

As for the next step, if I was doing the work, I would omit objects that can't be built from source, like ACU, and simply use the binary from the distribution disks. That make not make sense for you, given that you want to make an official release.

I don't. I'm only interested in studying and improving the source that is available. I would create two directories of pure source that CAN be built, 501 and 502, get both working, and then freeze them forever. If I found something broken in 502, I would check 501 too see if it's broken there too. If not, look at the 501 -> 502 diff to see what file patch or patches caused the breakage. If broken in both, then debug 502, find the cause and fix it. I would not call 502 "latest" because to me that implies ongoing development. I would use a third name for that.

But that kind of work is heavy lifting, which I avoid. You are doing the work, so it's up to you.

**Comment by src153** [comment]
I downloaded github desktop for Windows. I guess I'll install it and see what it's all about. Is that what you use? We're proposing a big reorganization. Is git well suited for big reorganizations? Would it be any better if we ran our own real git server instead of using github? Just wondering.

**Comment by the-grue** [comment]
I just use the github web interface and command line git on my Linux devices.  Started playing around with VSCode for an IDE, but it isn't suited to this project.

**Comment by src153** [comment]
I didn't want to learn git, but looks like I must, to save time. To answer my own question about file dates, I bought a $10 per month VPS, and used it to cloned the pcmos github repo. I see the file dates are today's date, and the DOS file names are uppercase. Yuck.

While you're working to reorganize, I will use my own repo, to pretty things up the way I like. It's not just cosmetic, I can work faster when I'm not distracted by things that annoy me.

If you want a user account on the VPS, for fun, or whatever, let me know. The provider does have some rules, no porn, no IRC unless approved, but that's about all.

**Comment by src153** [comment]
While looking for the cumulative access bitmap, I noticed the is a kernel1,mak. Except for whitespace, it's identical to kernel.mak. I don't know about you, but I get rid of any extra junk I can.

**Comment by the-grue** [comment]
Kernel1.mak didn't make it over in my cleanup.  Just kernel.mak, makeutil.mak, and maketerm.mak.

**Comment by the-grue** [comment]
These audits have been merged into the repo.  I will close this issue.
